scripts/checks.sh

```diff
@@ -46,7 +49,105 @@ function log_status()
 	jq -cn --argjson g "${1}" --arg f "${2}" --arg s "${3}" '[{"name":$f,"success":$g,"status":$s}]'
 }
 
+function test_upstream_dns()
+{
+	# force dnsmasq to print statistics to journal to read them back
+	systemctl kill -s USR1 dnsmasq
+	for i in $(journalctl -u dnsmasq | awk 'match($0, /[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]\.[12]?[0-9]?[0-9]/) {print substr($0, RSTART, RLENGTH)}'| sort -u); do
+		# shellcheck disable=SC2001
+		for j in "${external_fqdn}" "$(echo "${API_ENDPOINT}" | sed -e 's@^http.://@@')"; do
+			if ! nslookup "${j}" "${i}" > /dev/null 2>&1; then
+				echo "${FUNCNAME[0]}: DNS lookup failed for ${j} via upstream: ${i}"
+			fi
+		done;
+	done
+}
+
+function test_wifi()
+{
+	local -i wifi_active_interfaces
+	wifi_active_interfaces=$(nmcli --fields TYPE,STATE device status | awk '/wifi *connected/')
+	if (( wifi_active_interfaces > 0 )); then
+		local -i wifi_strength
+		wifi_strength=$(nmcli -f IN-USE,SIGNAL device wifi list | awk '/^\*/{print $2}')
+		if (( wifi_strength < wifi_threshold )); then
```

Comment:
I've never seen `(())` before, what's special about that vs `[[]]`/`[]`?

Reply:
`(())` is used to do math: http://faculty.salina.k-state.edu/tim/unix_sg/bash/math.html#bash-arithmetic